pkg/controller/replication/replication_controller.go

```diff
@@ -214,7 +214,7 @@ func (rm *ReplicationManager) updateRC(old, cur interface{}) {
 	// this function), but in general extra resyncs shouldn't be
 	// that bad as rcs that haven't met expectations yet won't
 	// sync, and all the listing is done using local stores.
-	if oldRC.Spec.Replicas != curRC.Spec.Replicas {
+	if oldRC.Spec.Replicas != nil && curRC.Spec.Replicas != nil && *(oldRC.Spec.Replicas) != *(curRC.Spec.Replicas) {
```

Comment:
Replicas is defaulted when nil so you can remove the nil comparisons
https://github.com/kubernetes/kubernetes/blob/cd37aaafb3b057f197225b5a92872aec60f054c8/pkg/api/v1/defaults.go#L63